src/main/java/seedu/algobase/model/plan/UniquePlanList.java

```diff
@@ -0,0 +1,138 @@
+package seedu.algobase.model.plan;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.algobase.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.algobase.model.plan.exceptions.DuplicatePlanException;
+import seedu.algobase.model.plan.exceptions.PlanNotFoundException;
+
+
+/**
+ * A list of plans that enforces uniqueness between its elements and does not allow nulls.
+ * A Plan is considered unique by comparing using {@code Plan#isSamePlan(Plan)}. As such, adding and
+ * updating of plans uses Plan#isSamePlan(Plan) for equality so as to ensure that the Plan being added
+ * or updated is unique in terms of identity in the UniquePlanList. However, the removal of a Plan uses
+ * Plan#equals(Object) so as to ensure that the Plan with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Plan#isSamePlan(Plan)
+ */
+public class UniquePlanList implements Iterable<Plan> {
```

Comment:
I don't think we need to enforce uniqueness on plans? @le0tan What do you think?

There is already an existing PlanList class anyway, so remove this class and modify the other one if necessary.